src/main/io/osd.c

```diff
@@ -1405,6 +1405,27 @@ static bool osdDrawSingleElement(uint8_t item)
             break;
         }
 
+    case OSD_EFFICIENCY:
+        {
+            // amperage is in centi amps, speed is in cms/s. We want
+            // mah/km. Values over 999 are considered useless and
+            // displayed as "---""
+            const int32_t maxEfficiencyValue = 999;
+            int32_t value = maxEfficiencyValue + 1;
+            if (gpsSol.groundSpeed > 0) {
+                value = (amperage / gpsSol.groundSpeed) / 0.0036f;
```

Comment:
You need explicit conversion to `float` here. `amperage` and `gpsSol.groundSpeed` are integer. For efficiency below 278 mah/km `amperage / gpsSol.groundSpeed` will yield zero.

Reply:
I hadn't realised... what if we multiply `amperage` by `1/0.0036` (277.77, so we either pick 277 or 278) so we can remove the floating point math?

Reply:
Yeah, you can change to `value = 278 * amperage / gpsSol.groundSpeed;` Less than 0.1% error is marginal.

Reply:
Ok, let's do that